### ISSUE TITLE
printprops displays matchdeclare properties incorrectly when value is a lambda

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1871]
Imported from SourceForge on 2024-07-05 10:19:06
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-02-13 19:59:54
Original: https://sourceforge.net/p/maxima/bugs/3607

---

Looks like `printprops` just appends a symbol to the value of the `matchdeclare` property. That's OK when the value is a function name, but it yields an incorrect expression when the value is a lambda expression. E.g. from the description for `printprops`:

~~~~
          (%i1) matchdeclare ([aa, bb, cc], atom, [dd, ee], integerp);
          (%o1)                         done
          (%i2) matchdeclare (ff, floatnump, gg, lambda ([x], x > 100));
          (%o2)                         done
          (%i3) propvars (matchdeclare);
          (%o3)             [aa, bb, cc, dd, ee, ff, gg]
          (%i4) printprops (ee, matchdeclare);
          (%o4)                    [integerp(ee)]
          (%i5) printprops (gg, matchdeclare);
          (%o5)              [lambda([x], x > 100, gg)]
          (%i6) printprops (all, matchdeclare);
          (%o6) [lambda([x], x > 100, gg), floatnump(ff), integerp(ee),
                                integerp(dd), atom(cc), atom(bb), atom(aa)]
~~~~

Here %o5 should be `[lambda([x], x > 100)(gg)]` or something like that.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:19:08
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-02-17 23:43:52
Original: https://sourceforge.net/p/maxima/bugs/3607/#a43a

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:19:12
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-02-17 23:43:52
Original: https://sourceforge.net/p/maxima/bugs/3607/#ee6c

---

Fixed by commit f40d185. Closing this report as fixed.
